Sources/XCRemoteCache/Network/ReplicatedRemotesNetworkClient.swift

```diff
@@ -39,6 +41,9 @@ class ReplicatedRemotesNetworkClient: RemoteNetworkClientImpl {
         let group = DispatchGroup()
         var results: [Result<Void, NetworkClientError>] = Array(repeating: .failure(.noResponse), count: urls.count)
         urls.enumerated().forEach { index, url in
+            if maxConcurrentRequests > 0 && index > 0 && index % maxConcurrentRequests == 0 {
```

Comment:
I'm trying to understand this line. So for example if `maxConcurrentRequests` is 5 and there are 100 urls. Would this mean that every 5th url would `wait()` and the rest would continue to `enter()`?

Reply:
>Would this mean that every 5th url would wait()
Yes, and that block others from `enter()` ( `wait()` will block the thread where we enumerate )

Reply:
Isn't that approach batching uploads into groups of `maxConcurrentRequests` rather than limiting number of concurrent requests? The `maxConcurrentRequests`th request will not be triggered until all already scheduled jobs (`0...(maxConcurrentRequests-1)`) call `leave()`?
Maybe we could call that parameter something with "xxxxx_batch_size"?